src/main/java/uk/gov/hmcts/ccd/auditlog/aop/AuditContextHolder.java

```diff
@@ -0,0 +1,21 @@
+package uk.gov.hmcts.ccd.auditlog.aop;
+
+public class AuditContextHolder {
+
+    private AuditContextHolder() {}
+
+    private static final InheritableThreadLocal<AuditContext> threadLocal = new InheritableThreadLocal<>();
```

Comment:
ThreadLocal are a bit tricky and could cause mem leaks. For my knowledge, why we need to use it here? what problem is this addressing?

Reply:
it depends on how you use it. how do you pass to Interceptor then..?